haas/api.py

```diff
@@ -274,7 +327,7 @@ def _have_attachment(nic, query):
     if nic.current_action:
         raise BlockedError("A networking operation is already active on the nic.")
 
-    if (network.access is not None) and (network.access is not project):
+    if (network.access) and (project not in network.access):
```

Comment:
Getting a fresh look at it I'm realizng they're not semantically equivalent anyway; without the first part it won't do the right thing with public networks. Nevermind.